velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -97,6 +98,52 @@ class DecimalBaseFunction : public exec::VectorFunction {
   const uint8_t bRescale_;
 };
 
+template <
+    typename R /* Result Type */,
+    typename A /* Argument */,
+    typename Operation /* Arithmetic operation */>
+class DecimalUnaryBaseFunction : public exec::VectorFunction {
+ public:
+  DecimalUnaryBaseFunction(uint8_t aRescale) : aRescale_(aRescale) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto rawResults = prepareResults(rows, resultType, context, result);
+    if (args[0]->isConstantEncoding()) {
```

Comment:
You need to decode the arguments (see `DecodedArgs`).

Reply:
For unary operations, the arguments are not encoded. Refer: https://github.com/facebookincubator/velox/pull/4557#discussion_r1167016474